docs/protocol/papers/blockchain-programs.md

```diff
@@ -209,20 +210,20 @@ When the program is instantiated with a `publicKey` value — to be used in an o
 When the output is spent and the control program is run:
 
 1. The virtual machine first takes arguments specified in the input witness and pushes them to the stack. In this case, that argument is a signature.
-2. The program then executes, first pushing the public key and then the transaction hash to the stack. 
-3. The public key and transaction hash are then swapped to put them in the correct order for the following `CHECKSIG` instruction. 
-4. `CHECKSIG` pops all three items off the stack to check the signature, pushing `true` or `false` to the stack. 
+2. The program then executes, first pushing the public key and then the transaction hash to the stack.
+3. The public key and transaction hash are then swapped to put them in the correct order for the following `CHECKSIG` instruction.
+4. `CHECKSIG` pops all three items off the stack to check the signature, pushing `true` or `false` to the stack.
 5. `VERIFY` then pops the top value from the stack and causes the program to fail if the value is `false`. (In an actual control program, the `VERIFY` instruction of the last condition in a path is omitted, since it is performed by the VM itself.)
 
 Many control, issuance, and consensus programs use a multisignature check.
 
-	program MultiKeyProgram(n, m, publicKeys[n]) {
-		path spend(signatures[m]) {
+	program MultiKeyProgram(n: Number, m: Number, publicKeys: PublicKey[n]) {
```

Comment:
While this gets "m of n" across, it seems a bit awkward to me as actual syntax and unclear how it would map to an actual type system.

The KISS alternative I'd suggest is to just pass an array, which implicitly carries with it an `n`, e.g.:

`program MultiKeyProgram(threshold: Number, publicKeys: PublicKey[])`

Beyond that you're trying to parameterize the array type with additional type information, which feels like something that can't be done cleanly without something like a dependent type system (or at least a weak form of one... [this particular problem comes up frequently in Rust](https://github.com/rust-lang/rfcs/pull/1657/files))